src/components/common/SelectTypes.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+
+import { Stack } from '@mui/material';
+import Checkbox from '@mui/material/Checkbox';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import ListItemText from '@mui/material/ListItemText';
+import MenuItem from '@mui/material/MenuItem';
+import OutlinedInput from '@mui/material/OutlinedInput';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+
+import { ItemType } from '@graasp/sdk';
+import { ItemIcon } from '@graasp/ui';
+
+import { useBuilderTranslation, useEnumsTranslation } from '@/config/i18n';
+import { BUILDER } from '@/langs/constants';
+
+import { useFilterItemsContext } from '../context/FilterItemsContext';
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+    },
+  },
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const { S3_FILE, ...BUILDER_ITEM_TYPES } = ItemType; // exclude S3 file
+
+export const SelectTypes = (): JSX.Element => {
+  const { itemTypes, setItemTypes } = useFilterItemsContext();
+  const { t: translateEnums } = useEnumsTranslation();
+  const { t: translateBuilder } = useBuilderTranslation();
+
+  const types = Object.values(BUILDER_ITEM_TYPES).sort((t1, t2) =>
+    translateEnums(t1).localeCompare(translateEnums(t2)),
+  );
+
+  const handleChange = (event: SelectChangeEvent<typeof itemTypes>) => {
+    const {
+      target: { value },
+    } = event;
+
+    if (typeof value === typeof itemTypes) {
```

Comment:
Isn't that always true? isn't `value` always a `string`?

Reply:
It was to try to avoid casting but it is still here, so I can remove the if.